modules/user/user.service.js

```diff
@@ -0,0 +1,155 @@
+const { AuthenticationError, UserInputError } = require('apollo-server');
+const bcrypt = require('bcryptjs');
+const User = require('./user.model');
+const {
+  validateRegisterInput,
+  validateLoginInput,
+} = require('../../utils/validateUser');
+const generateToken = require('../../utils/createToken');
+
+class UserService {
+  getAllUsers() {
+    return User.find();
+  }
+
+  getUserById(id, auth) {
+    if (auth.errors) throw new Error(auth.errors);
+    return User.findById(id);
+  }
+
+  async updateUser(id, {
+    firstName, lastName, email, password,
+  }, auth) {
+    if (auth.errors) throw new Error(auth.errors);
+
+    const checkedUser = await User.findOne({ email });
+
+    if (!checkedUser) {
+      const massage = 'User with provided email not found';
+      throw new UserInputError(massage, {
+        errors: {
+          email: massage,
+        },
+      });
+    }
+
+    const { errors } = await validateRegisterInput.validateAsync({
```

Comment:
You should validate input data before reaching the database

Reply:
fixed